Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -615,15 +620,54 @@ TEST_F(SerializerTest, BadTag) {
   // Specifically 31. 0xf8 represents field number 31 encoded as a varint.
   Mutate(&bytes[0], /*expected_initial_value=*/0x58, /*new_value=*/0xf8);
 
-  // TODO(rsgowman): The behaviour is *temporarily* slightly different during
-  // development; this will cause a failed assertion rather than a failed
-  // status. Remove this EXPECT_ANY_THROW statement (and reenable the
-  // following commented out statement) once the corresponding assert has been
-  // removed from serializer.cc.
-  EXPECT_ANY_THROW(ExpectFailedStatusDuringFieldValueDecode(
-      Status(FirestoreErrorCode::DataLoss, "ignored"), bytes));
-  // ExpectFailedStatusDuringFieldValueDecode(
-  //    Status(FirestoreErrorCode::DataLoss, "ignored"), bytes);
+  ExpectFailedStatusDuringFieldValueDecode(
+      Status(FirestoreErrorCode::DataLoss, "ignored"), bytes);
+}
+
+TEST_F(SerializerTest, BadFieldValueTagWithOtherValidTagsPresent) {
+  // A bad tag should be ignored, in which case, we should successfully
+  // deserialize the rest of the bytes as if it wasn't there. To craft these
+  // bytes, we'll use the same technique as
+  // EncodesFieldValuesWithRepeatedEntries (so go read the comments there
+  // first.)
+
+  // Copy of the real one (from the nanopb generated document.pb.h), but with
+  // only boolean_value and integer_value.
+  struct google_firestore_v1beta1_Value_Fake {
+    bool boolean_value;
+    int64_t integer_value;
+  };
+
+  // Copy of the real one (from the nanopb generated document.pb.c), but with
+  // only boolean_value and integer_value. Also modified such that integer_value
```

Comment:
Entirely optional: using a named variable could make the latter part of the comment unnecessary:
```cpp
const int invalid_tag = 31;
// ...
PB_FIELD(invalid_tag, ...
```

Reply:
Done, though I ended up keeping most of the comment.